### ISSUE TITLE
[Support]: Cannot run 64vit version of Frigate on RPi4 8Gig

### ISSUE THREAD

**Issue by LordNex** [blakeblackshear/frigate#2588]
### Describe the problem you are having

I just received my Coral TPU so I reinstalled my Raspberry Pi 4 8 gig with the lite buster 64bit. Then walked through the install. I did run into some issues getting Docker-Compose installed, but eventually was able to with pip instead of apt. I tried both the stable:aarch64 and frigate:0.10.0-beta6-aarch64 as listed on [https://github.com/blakeblackshear/frigate/releases](url)

### Version

NA

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
pi@POWFRIGATE:~/docker-compose/frigate $ sudo docker-compose up                                                                                                         Creating network "frigate_default" with the default driver
Pulling frigate (blakeblackshear/frigate:stable-aarch64)...
stable-aarch64: Pulling from blakeblackshear/frigate
ab2d02b1ec42: Pull complete
fe11f32baebb: Pull complete
4fbd7490df0b: Pull complete
5e56ae8abb89: Pull complete
cd106601db74: Pull complete
b7a58820f01f: Pull complete
b7bd8a0601c5: Pull complete
40d9081310e2: Pull complete
3e16bb824a7b: Pull complete
99477d17cbba: Pull complete
7390bfbffd14: Pull complete
631bc3ef97f3: Pull complete
fa850c534d62: Pull complete
a54db136ae44: Pull complete
d34e0c02b037: Pull complete
a884c10e90fa: Pull complete
1c68cbb5fe58: Pull complete
Digest: sha256:144b15112eb17d3f7ade840310ac31ef42413abd7839551d1ea3d9754f583b73
Status: Downloaded newer image for blakeblackshear/frigate:stable-aarch64
Creating frigate ... done
Attaching to frigate
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
pi@POWFRIGATE:~/docker-compose/frigate $ ^C
pi@POWFRIGATE:~/docker-compose/frigate $ sudo docker-compose up
frigate is up-to-date
Attaching to frigate
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate exited with code 1
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
pi@POWFRIGATE:~/docker-compose/frigate $ sudo nano ~/docker-compose/frigate/docker-compose.yml
pi@POWFRIGATE:~/docker-compose/frigate $ sudo docker-compose up                                                                                                         Pulling frigate (blakeblackshear/frigate:0.10.0-beta6-aarch64)...
0.10.0-beta6-aarch64: Pulling from blakeblackshear/frigate
a39c84e173f0: Pull complete
8c43aaf426d4: Pull complete
19ceb23a2d58: Pull complete
11cb80249db0: Pull complete
e91ef4ccf3af: Pull complete
9b3f9514083d: Pull complete
b7bd8a0601c5: Pull complete
ef052bdf05fd: Pull complete
5216cf5b9c02: Pull complete
0360a5aa154a: Pull complete
efa5acd991f2: Pull complete
2bb599552015: Pull complete
0a77732d1ffb: Pull complete
cc373490e164: Pull complete
8d901965bc43: Pull complete
b41861c1d0cb: Pull complete
e20cf3ebaf43: Pull complete
Digest: sha256:5e80a9287ee49fba422c8178e5dcf71c904815f74eadd88126d0587e88701f3b
Status: Downloaded newer image for blakeblackshear/frigate:0.10.0-beta6-aarch64
Recreating frigate ... done
Attaching to frigate
frigate    | standard_init_linux.go:228: exec user process caused: exec format error
frigate exited with code 1
```

### FFprobe output from your camera

```shell
NA
```

### Frigate stats

```json
NA
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Amcrest AD-410, Wyze Pan Cam, Reolink RCL-520A

### Any other information that may be helpful

I previously was able to run just the CPU version of Arm7 before I received the Coral. Once I did I wanted to wipe clean 
 and make sure I had everything set for 64 bit so i can utilize all 8 gigs.

**Comment by blakeblackshear** [comment]
What is the output of `uname -m`?

**Comment by LordNex** [comment]
Armv71

Could have sworn I installed the 64 bit version of the ISO? Is there a reason it would default to 32 bit? Probably because no other RPi is?

Benjamin T Powers

On Jan 5, 2022, at 6:46 AM, Blake Blackshear ***@***.***> wrote:

﻿

What is the output of uname -m?

**Comment by blakeblackshear** [comment]
Not sure what could have happened. I have successfully installed 64 bit, so I know it works.

**Comment by LordNex** [comment]
I thought I remember reading somewhere that you have to modify a file to force the Pi to boot into 64 bit mode. Either way with that TPU I’m getting about 20ms for all 5 streams. Still tweaking them a bit as well but I really want to get this into fully 64 bit mode. I’ll dig around and see if I can find it. I’d you happen to remember rbbbb Yea I’ll get by

Benjamin T Powers

On Jan 5, 2022, at 7:25 PM, Blake Blackshear ***@***.***> wrote:

﻿

Not sure w com have on a few more days as a hat could have happened. I have successfully ihhg G G vbvv G nstalled 64 bit, so I know it works.

**Comment by LordNex** [comment]
@blakeblackshear  I found out that by default the 64bit version of the os is not enabled on the raspian image. I had to modify the /boot/config.txt and add arm_64bit=1 to the end and reboot. Once I did that I was able to run the aarch version of Frigate. Might be worth throwing a blurb into the install read me as I downloaded the latest image. You'd think they would have that enabled for that specific version of the kernel
